enumerable.rb

```diff
@@ -0,0 +1,109 @@
+module Enumerable
+  def my_each
+    # Should return the array except block is given
+    return to_enum unless block_given?
+
+    each do |i|
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    # Should return the array except block is given
+    return to_enum unless block_given?
+
+    (0..size - 1).my_each do |i|
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    # an empty array to store our selected element
+    arr = []
+    (0...size - 1).my_each do |i|
+      arr << self[i] if yield(self[i])
+    end
+    arr
+  end
+
+  def my_all?
+    status = true
+    my_each do |x|
+      status = false if x.nil? || x == false
+    end
+    return status unless block_given?
+
+    count = 0
+    size.times do |i|
+      count += 1 unless yield self[i]
+    end
+    count.zero?
+  end
```

Comment:
#### my_all? 

- [ ] when a class is passed as an argument returns true only if all of the collection is a member of such class, otherwise, it should return false

     > [1, 2.5, 'a', 9].my_all?(Integer)  ==  [1, 2.5, 'a', 9].all?(Integer) => false

- [ ] when a Regex is passed as an argument returns true if all of the collection matches the Regex

     >  %w[dog door rod blade].my_all?(/d/) == %w[dog door rod blade].all?(/d/) => true

- [ ] when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

     > [3,4,7,11].my_all?(3) == [3,4,7,11].all?(3)  => false